src/containers/course-section/CourseSectionContainer.tsx

```diff
@@ -144,6 +144,30 @@ const CourseSectionContainer: FC<SectionProps> = ({
     setDescriptionInput(event.target.value)
   }
 
+  useEffect(() => {
+    const sectionIndex = sections.findIndex(
+      (section) => section.id === sectionData.id
+    )
+    let sectionValue = sections[sectionIndex]
+
+    sectionValue = {
+      ...sectionValue,
+      title: titleInput,
+      description: descriptionInput,
+      lessons: lessons,
+      quizzes: quizzes,
+      attachments: attachments
+    }
+
+    const newSectionsArray = [
+      ...sections.slice(0, sectionIndex),
+      sectionValue,
+      ...sections.slice(sectionIndex + 1, sections.length)
+    ]
```

Comment:
Have you tested the performance of the code? Wouldn't that slow it down with 10 sections or more? Is there a problem with the sections getting mixed up when you make changes or is that all right?

Reply:
Yes, I tested it. There are no problems if we make changes because it moves changes to the sections simultaneously when we change. But yes, it slows down a little when it is 10 sections.

Reply:
> Yes, I tested it. There are no problems if we make changes because it moves changes to the sections simultaneously when we change. But yes, it slows down a little when it is 10 sections.

please create task for refactoring later